nussl/separation/duet.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import axes3d
+from scipy import signal
+
+import nussl.audio_signal
+import nussl.constants
+import nussl.spectral_utils
+import nussl.utils
+import mask_separation_base
+import masks
+
+
+class Duet(mask_separation_base.MaskSeparationBase):
+
+    """Implements the Degenerate Unmixing Estimation Technique (DUET) algorithm.
```

Comment:
Overall this documentation is a lot nicer. Good job beefing it up. One thing I think that could help the docs a lot is if you gave a brief overview about how the algorithm works (feel free to use math and/or code examples from this file). Just to add context to what 'attenuation' and 'delay' are. Add a paragraph or so, not too much so that a user gets the gist and can make intelligent decisions about how to set the parameters.